src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -364,6 +357,8 @@ public ISurrogated FromSurrogate(ActorSystem system)
 
         public override ISurrogate ToSurrogate(ActorSystem system)
         {
+            if (Decider is LocalOnlyDecider)
+                throw new NotSupportedException("Can not serialize LocalOnlyDecider");
```

Comment:
This line prevents objects referring to a local only decider from being serialized.
Json.NET silently ignored this and passed the message without the decider, potentially leading to incorrect error handling on the remote system.